app/api/v1/auth/index.js

```diff
@@ -2,200 +2,145 @@ const express = require('express');
-		Activity.accountRequestedResetPassword(user.uuid);
-	}).catch(next);
-});
-
-/**
- * Reset the password for a user
- * 
- * Given an access code and the new password for a user, change it
- * The POST body should have a 'user' object in the form { newPassword, confPassword }
+ * On success, this route will return the user's public profile and a user token containing user's ID and privilege levels
  */
-router.post('/resetPassword/:accessCode', (req, res, next) => {
-	if (!req.params.accessCode)
-		return next(new error.BadRequest('Invalid access code'));
-	if (!req.body.user || !req.body.user.newPassword || !req.body.user.confPassword)
-		return next(new error.BadRequest('Invalid user data'));
-	if (req.body.user.newPassword !== req.body.user.confPassword)
-			return next(new error.UserError('Passwords do not match'));
-	if (req.body.user.newPassword.length < 10)
-			return next(new error.UserError('New password must be at least 10 characters'));
-
-	// find the user using the given access code
-	User.findByAccessCode(req.params.accessCode).then(user => {
-		// if no such user was found, probably the access code is invalid or non-existent
-		if (!user || !user.requestedPasswordReset())
-			throw new error.BadRequest('Invalid access code');
-		// use the new password to update the user's hash and account state
-		return User.generateHash(req.body.user.newPassword).then(hash => {
-			user.hash = hash;
-			user.state = 'ACTIVE';
-			return user.save();
-		});
-	}).then(user => {
-		res.json({ error: null });
-		// record that the user reset their password
-		Activity.accountResetPassword(user.uuid);
-	}).catch(next);
+router.post('/login', (req, res, next) => {
+  if (!req.body.tokenId || req.body.tokenId.length < 1) return next(new error.BadRequest('Invalid token.'));
+
+  const createUserToken = (user) => {
+    // create a token with the user's ID and privilege level
+    jwt.sign(
+      {
+        uuid: user.getDataValue('uuid'),
+        admin: user.isAdmin(),
+        registered: !user.isPending(),
+      },
+      config.session.secret,
+      { expiresIn: TOKEN_EXPIRES },
+      (err, token) => {
+      if (err) return next(err);
+    
+      // respond with the token upon successful login
+      res.json({
+        error: null,
+        user: user.getPublicProfile(),
+        token,
+      });
+      // record that the user logged in
+      Activity.accountLoggedIn(user.uuid);
+      },
+    );
+  };
+
+  client
+    .verifyIdToken({
+      idToken: req.body.tokenId,
+      audience: config.google.clientId,
+    })
+    .then((ticket) => {
+      const { email } = ticket.getPayload();
+
+      if (!email.toLowerCase().endsWith(config.google.hostedDomain))
+        return next(new error.Unauthorized('Unauthorized email'));
+
+      User.findByEmail(email.toLowerCase())
+        .then(user => ({ user, ticket }))
+        .then((userData) => {
+          const { user, ticket } = userData;
```

Comment:
A bit confused by this first `.then` in this chain - why can we not do 
```suggestion
        .then(({ user, ticket }) => {
```

Reply:
Done.